src/lib.rs

```diff
@@ -0,0 +1,10 @@
+mod common;
+
+#[cfg(target_os = "linux")]
+mod linux;
+
+#[cfg(target_os = "linux")]
+#[ctor::ctor]
```

Comment:
Well my thought was that the injector gets just `LD_PRELOAD`ed when factorio starts, gets librivets and loads it. \
Means there only needs to be such a special executable on windows since that requires special injection stuff.